packages/nodes-base/nodes/Code/JsTaskRunnerSandbox.ts

```diff
@@ -39,24 +40,37 @@ export class JsTaskRunnerSandbox {
 			: this.throwExecutionError(executionResult.error);
 	}
 
-	async runCodeForEachItem(): Promise<INodeExecutionData[]> {
+	async runCodeForEachItem(numInputItems: number): Promise<INodeExecutionData[]> {
 		validateNoDisallowedMethodsInRunForEach(this.jsCode, 0);
+
 		const itemIndex = 0;
+		const chunks = this.chunkInputItems(numInputItems);
+		let executionResults: INodeExecutionData[] = [];
 
-		const executionResult = await this.executeFunctions.startJob<INodeExecutionData[]>(
-			'javascript',
-			{
-				code: this.jsCode,
-				nodeMode: this.nodeMode,
-				workflowMode: this.workflowMode,
-				continueOnFail: this.executeFunctions.continueOnFail(),
-			},
-			itemIndex,
-		);
+		for (const chunk of chunks) {
+			const executionResult = await this.executeFunctions.startJob<INodeExecutionData[]>(
+				'javascript',
+				{
+					code: this.jsCode,
+					nodeMode: this.nodeMode,
+					workflowMode: this.workflowMode,
+					continueOnFail: this.executeFunctions.continueOnFail(),
+					chunk: {
+						startIndex: chunk.startIdx,
+						count: chunk.count,
+					},
+				},
+				itemIndex,
+			);
 
-		return executionResult.ok
-			? executionResult.result
-			: this.throwExecutionError(executionResult.error);
+			if (!executionResult.ok) {
+				return this.throwExecutionError(executionResult.error);
+			}
+
+			executionResults = executionResults.concat(executionResult.result);
```

Comment:
Should we `push` instead of creating a new array each iteration?

Reply:
With `push` we'd have to iterate over each item. With concat we don't have to do that